From 328fbd74f87c267baa5b21f8c3561c9b0954e1db Mon Sep 17 00:00:00 2001
From: zhangyichix <yichix.zhang@intel.com>
Date: Wed, 26 Oct 2022 06:05:04 +0000
Subject: [PATCH] Modify the default value of the bitrate mode

case:
android.mediav2.cts.EncodeDecodeAccuracyTest#testEncodeDecodeAccuracyRGB

With setting the default of bitrate mode to BITRATE_CONST and it will be not
changed in the format by user, MediaCodecInfo.CodecCapabilities.isFormatSupported
 will return false.

Tracked-On: OAM-104455
Signed-off-by: zhangyichix <yichix.zhang@intel.com>
Signed-off-by: Kothapeta, BikshapathiX <bikshapathix.kothapeta@intel.com>
---
 .../src/mfx_c2_encoder_component.cpp          | 22 ++++++++++++++++++-
 1 file changed, 21 insertions(+), 1 deletion(-)

diff --git a/c2_components/src/mfx_c2_encoder_component.cpp b/c2_components/src/mfx_c2_encoder_component.cpp
index 64905e3..19f8815 100755
--- a/c2_components/src/mfx_c2_encoder_component.cpp
+++ b/c2_components/src/mfx_c2_encoder_component.cpp
@@ -214,7 +214,7 @@ MfxC2EncoderComponent::MfxC2EncoderComponent(const C2String name, const CreateCo
         AllocUniqueString<C2PortMediaTypeSetting::input>("video/raw"));
 
     pr.AddValue(C2_PARAMKEY_BITRATE_MODE,
-        std::make_unique<C2StreamBitrateModeTuning::output>(SINGLE_STREAM_ID, C2Config::bitrate_mode_t::BITRATE_CONST));
+        std::make_unique<C2StreamBitrateModeTuning::output>(SINGLE_STREAM_ID, C2Config::bitrate_mode_t::BITRATE_VARIABLE));
 
     pr.AddValue(C2_PARAMKEY_REQUEST_SYNC_FRAME, 
         std::make_unique<C2StreamRequestSyncFrameTuning::output>(SINGLE_STREAM_ID, false));
@@ -1591,6 +1591,26 @@ c2_status_t MfxC2EncoderComponent::QueryParam(const mfxVideoParam* src, C2Param:
                 }
                 break;
             }
+            case kParamIndexBitrateMode: {
+                if (nullptr == *dst) {
+                    *dst = new C2StreamBitrateModeTuning::output();
+                }
+
+                C2StreamBitrateModeTuning::output* setting = static_cast<C2StreamBitrateModeTuning::output*>(*dst);
+                switch (m_mfxVideoParamsConfig.mfx.RateControlMethod) {
+                    case MFX_RATECONTROL_CBR:
+                        setting->value = C2Config::bitrate_mode_t::BITRATE_CONST;
+                        break;
+                    case MFX_RATECONTROL_VBR:
+                        setting->value = C2Config::bitrate_mode_t::BITRATE_VARIABLE;
+                        break;
+                    default:
+                        setting->value = C2Config::bitrate_mode_t::BITRATE_IGNORE;
+                        break;
+                }
+                MFX_DEBUG_TRACE_STREAM("QueryParam BitrateMode = " << setting->value);
+                break;
+            }
             default:
                 res = C2_BAD_INDEX;
                 break;
-- 
2.38.1

